dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -147,7 +178,113 @@ public Field save(final Field field, final User user) throws DotDataException, D
       return result;
 	}
 
-  @WrapInTransaction
+    /**
+     * Verify if a relationship already exists for this field. Otherwise, a new relationship object
+     * is created
+     * @param field
+     * @param contentTypeAPI
+     * @param type
+     * @return
+     * @throws DotDataException
+     */
+    @VisibleForTesting
+    Optional<Relationship> getRelationshipForField(Field field, ContentTypeAPI contentTypeAPI,
+            ContentType type) throws DotDataException {
+        Relationship relationship;
+        ContentType relatedContentType;
+        try {
+            final String[] relationType = field.relationType().split("\\.");
+            final int cardinality = Integer.parseInt(field.values());
+
+            //check if cardinality is valid
+            if (cardinality < 0 || cardinality >= RELATIONSHIP_CARDINALITY.values().length){
+                throw new DotDataException("Cardinality value is incorrect");
+            }
+
+            //we need to find the id of the related structure using the velocityVarName set in the relationType
+            try {
+                relatedContentType = contentTypeAPI.find(relationType[0]);
+            } catch (NotFoundInDbException e) {
+                Logger.debug(this, "No relationship has been saved for field " + field.name()
```

Comment:
I think a debug level is better here because at the end of the PP process all relationships will be saved, except in those cases where a problem arises in the relationship itself, not the content type, in which case I log an error.